### ISSUE TITLE
CERTIFICATE_VERIFT_FAILED

### ISSUE THREAD

**Issue by UniqHu** [Enough-Software/enough_mail#167]
Use smtp for xoauth2  mechanism. But got handshakeexception.OS error:CERTIFICATE_VERIFT_FAILED:application verification failure.   However i don't need verify certificate.  How to skip that

**Comment by robert-virkus** [comment]
Can you explain your use case a bit more? For me it sounds dangerous to skip the SSL verification...

I would be possible to expose a ` bool Function(X509Certificate)? onBadCertificate` callback, so that you can handle this case yourself:

```
[onBadCertificate] is an optional handler for unverifiable certificates.
The handler receives the [X509Certificate], and can inspect it and decide 
(or let the user decide) whether to accept the connection or not. 
The handler should return true to continue the [SecureSocket] connection.
```

**Comment by robert-virkus** [comment]
released in v1.3.5, you can register the callback in the constructors of `MailClient`, `SmtpClient`, `ImapClient` and `PopClient`.
